### DISCUSSION TITLE
How can we add a custom commit id generator?

### DISCUSSION THREAD

**Discussion by Anupam2528** [javers/javers#1363] [Q&A]
Hi Team,
I want to add a custom commit id generator in the javers enabled application.
I saw in JaversBuilder we have a method as withCustomCommitIdGenerator, however it is not a public one.
So just want to know if we have any way to do this.

Thank you!!

**Comment by bartoszwalacik** [comment]
Hi, `withCustomCommitIdGenerator()` is used internally in some tests, don't see any reasons why not to make it public. We will accept your PR, please link this discussions in your PR
